### PR TITLE
Add tests and configuration capacity for SVML.

### PR DESCRIPTION
**Pull request by stuartarchibald**
Adds:
 * Runtime based detection of SVML.
 * Tests for SVML for both standard and fastmath modes.
 * Adds the capacity to disable the use of SVML via an environment
   variable.
 * Tests the above.

### CODE REVIEW

File: numba/tests/test_svml.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        self.check(math_sin_scalar, 7., fast_pattern=pat)
+
+    def test_svml(self):
+        # loops both with and without fastmath should use SVML.
+        # The high accuracy routines are dropped if `fastmath` is set
+        std = "$__svml_sin8_ha,"
+        fast = "$__svml_sin8,"  # No `_ha`!
+        self.check(math_sin_loop, 10, std_pattern=std, fast_pattern=fast)
+
+    def test_svml_disabled(self):
```
Comment (stuartarchibald) [inline numba/tests/test_svml.py:125]: I probably ought to add a note to this test explaining why its in a subprocess. It's due to Numba picking up SVML at init time and permanently loading a library, this making the standard configuration override not work.

File: numba/__init__.py [hunk 2]
```diff
@@ -119,10 +104,41 @@ def _ensure_pynumpy():
...
 _ensure_pynumpy()
 
+# we know llvmlite is working as the above tests passed, import it now as SVML
+# needs to mutate runtime options (sets the `-vector-library`).
+import llvmlite
+
+"""
+Is set to True if Intel SVML is in use.
+"""
+USING_SVML = _try_enable_svml()
```
Comment (stuartarchibald) [inline numba/__init__.py:141]: I'm unsure as to whether this ought to be here.

**Review by anton-malakhov — APPROVED** [review]
Thank you, Stuart! I see there are parts of the puzzle I was struggling with. Though it misses AVX2, prange, numpy ufuncs, and tests for all the other functions besides `sin()` but if this test is enough for promotion, I'll be happy to keep working on the rest of the tests in background while SVML will be already enabled.

**Comment by stuartarchibald** [comment]
No problem :)

What did you have in mind for AVX2, prange, ufuncs etc?

**Comment by anton-malakhov** [comment]
I have some draft of a test which reuses test_ufuncs and redefines compile() in order to add checks for the expected signatures. It could potentially be merged back into test_ufuncs and extended to test_parfors to do the same checks over all the forms of ufuncs and explicit loops. I still don't know how effectively get current cpu target in numba so that I'd avoid duplicated compilations.. but it looks like you were not concerned with that here.
The check itself could do more than just two modes, extending it to AVX2 and in the future to enhanced performance (EP) mode of SVML (which adds "superfast" mode to numba :) )

**Comment by stuartarchibald** [comment]
By signatures, do you mean the functions being called for e.g. transcendentals?

Regarding current CPU target, would the llvm-triple be suitable? Numba delegates CPU and CPU feature identification to LLVM via llvmlite:
```
In [1]: import llvmlite.binding as ll

In [2]: ll.get_default_triple()
Out[2]: 'x86_64-unknown-linux-gnu'

In [3]: ll.get_host_cpu_name()
Out[3]: 'core2'

In [4]: ll.get_host_cpu_features()
Out[4]: 
{'adx': False,
 'aes': False,
 'avx': False,
 'avx2': False,
... etc
```

Spoofing the CPU can be done with https://github.com/anton-malakhov/numba/pull/1/files#diff-0a268a00f049cf3901afe532e1725618R150, I think this would probably be the thing to use to ensure the presence of a known CPU/feature set when compiling, after which the assembler can be parsed to ensure that e.g. an AVX2 variant was used (`zmm` access?).

**Comment by anton-malakhov** [comment]
yes, I mean basically all the patterns expected from a SIMD code - using registers, packed vector instructions, and SVML calls. AVX2 target will produce e.g. `__svml_sin4` instead of `__svml_sin8`, where the number indicates the vector length.
Thanks! `cpu_name` and `cpu_features` are promising candidates.

**Pull request merged**